google-cloud-pubsub/src/main/java/com/google/cloud/pubsub/v1/Publisher.java

```diff
@@ -1044,7 +1049,7 @@ private OutstandingBatch popOutstandingBatch() {
 
     private void reset() {
       messages = new LinkedList<>();
-      batchedBytes = 0;
+      batchedBytes = topicNameSize;
```

Comment:
nit: Pass in `topicNameSize` as a parameter to the `MessagesBatch` constructor. You can add a `topicNameSize` field in the class so that this value can be used both here in `reset()` and above in `popOutstandingBatch()`.